src/challengepaths/2015/CommunityService.ts

```diff
@@ -0,0 +1,375 @@
+   * @param id The id the game HTML uses to identify the test; this is used primarily in runChoice.
+   * @param property The name of the test as a string, often used as part of the string property "csServicesPerformed".
+   * @param predictor A function that returns an estimate for the number of turns that the test will take given your character's current state.
+   * @param maximizeRequirements A Requirement object, if applicable, that aligns with the things needed to maximize for this particular test.
+   */
+  constructor(
+    id: number,
+    property: string,
+    predictor: () => number,
+    maximizeRequirements: Requirement | null = null
+  ) {
+    this.choice = id;
+    this.property = property;
+    this.predictor = predictor;
+    this.maximizeRequirements = maximizeRequirements;
+  }
+
+  /**
+   * @returns The id number of the test, used primarily in runChoice.
+   */
+  get id(): number {
+    return this.choice;
+  }
+  /**
+   * @returns The name of the test, used primarily as part of the string property "csServicesPerformed"
+   */
+  get name(): string {
+    return this.property;
+  }
+  /**
+   *  @returns The predicted number of turns this test will take given your character's current state.
+   */
+  get prediction(): number {
+    return this.predictor();
+  }
+  /**
+   * @returns A Requirement object, if applicable, that aligns with the things needed to maximize for this particular test.
+   */
+  get requirement(): Requirement | null {
+    return this.maximizeRequirements;
+  }
+
+  /**
+   * Checks the "csServicesPerformed" property to see whether mafia currently believes this test is complete.
+   * @returns Whether mafia currently believes this test is complete.
+   */
+  isDone(): boolean {
+    return get("csServicesPerformed").includes(this.property);
+  }
+
+  /**
+   * Maximizes based on the Requirement associated with this particular test.
+   */
+  maximize(): void {
+    if (this.maximizeRequirements) this.maximizeRequirements.maximize();
+  }
+
+  /**
+   * Attempts to turn in the test to the Council of Loathing.
+   * @returns Whether mafia believes the test is complete at the end of this function.
+   */
+  do(): boolean {
+    visitUrl("council.php");
+    runChoice(this.choice);
+    return this.isDone();
+  }
+
+  /**
+   * Wrapper function that prepares for a test and then completes it, adding time and turn details to the log.
+   * @param prepare A function that does all necessary preparations for this CS test, including choosing your outfit.
+   * @param beCertain Whether we should check council.php instead of mafia to determine whether the test is complete.
+   * @returns The output of the prepare function given, or null if the test is already complete.
+   */
+  run<T>(prepare: () => T, beCertain = false): T | null {
+    const finishedFunction = beCertain ? this.verifyIsDone : this.isDone;
+    if (finishedFunction()) return null;
+
+    print();
```

Comment:
No printing

Reply:
can I print straight to the log?

Reply:
or should this be the job of the scriptwriter with an `extends` or whatever to wrap the existing libram class?

Reply:
I have for now excised them, and worst case scenario people can create an extends class that wraps it, and override this method with something that wraps super.

Reply:
Best case scenario you and gausie fight, and in doing so, learn to understand the other's perspective and grow closer as friends and comrades